File: backend/app/schemas/invoice.py
import uuid
from typing import Optional
from pydantic import BaseModel


class InvoiceResponse(BaseModel):
    id: uuid.UUID
    invoice_number: str
    order_id: uuid.UUID
    subtotal: float
    delivery_fee: float
    tax_amount: float
    discount_amount: float
    total_amount: float
    currency: str
    payment_status: str
    pdf_url: Optional[str] = None

    model_config = {"from_attributes": True}

File: frontend/src/types/invoice.ts
export interface InvoiceLineItem {
  name: string;
  unit_price: number;
  quantity: number;
  line_total: number;
}

export interface Invoice {
  id: string;
  invoice_number: string;
  order_id: string;
  subtotal: number;
  delivery_fee: number;
  tax_amount: number;
  discount_amount: number;
  total_amount: number;
  currency: string;
  payment_status: string;
  pdf_url?: string | null;
}
